### DISCUSSION TITLE
Printhead hitting frame during probe.

### DISCUSSION THREAD

**Discussion by elkhunt** [Jyers/Marlin#1329] [Q&A] [answered]
I have an Ender3 v2 with ABL Probe.  When probing to do the mesh level the print head is hitting the z axis support leg by the control panel.  I have a cable chain installed and I believe my issue has to do with line 1368 in the configuration code.  I think if the +15 were taken out of the  #define X_MAX_POS (X_BED_SIZE + X_MIN_POS)  it would quit slamming into the frame.    I was able to open the code Visual Studio Code but I have not a clue as to what I'm doing and no Idea how to compile and reinstall this on the printer if I take that bit of code out to try and fix it for my individual printer.  Can anyone help me out or point me in the right direction?   So I can try and trouble shoot this?  Or does anyone else have any other ideas??

**Comment by hitboxr** [comment, marked as answer]
Should be a simple fix, although you may lose probe points by reducing your max X position. Install PlatformIO IDE (an extension for VSCode), which will allow you to one-click build this firmware. Open the project in PlatformIO by opening the extension from the panel on the left, and then clicking `Open Project` and navigating to the folder containing the `platformio.ini` file (the outermost folder named `Marlin`).

From the `Configuration Files\E3V2 Templates` folder, copy two configuration files that match your preferred bed level settings to the `Marlin\Marlin` subfolder. These should replace the two configuration files already there. Then, edit the line containing the problematic offset (keep in mind that reducing this will reduce the number of probe points that your printer will be able to reach). At the bottom of the screen, click the check mark button to build the firmware in PlatformIO. The resulting .bin file will be located in `.pio\build\STM32F103RET6_creality\`. Copy this onto your SD card and flash like you would any other firmware and you should be good to go.

**Comment by elkhunt** [comment]
Thank you so much!  That was not nearly as daunting as I thought it was going to be!  But my god after seeing what goes into making firmware Jyers has earned my respect holy cow that is a lot of work………
